themes-default/slim/webpack.config.js

```diff
@@ -5,6 +5,7 @@ const VueLoaderPlugin = require('vue-loader/lib/plugin');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const FileManagerPlugin = require('filemanager-webpack-plugin');
 const CopyWebpackPlugin = require('copy-webpack-plugin');
+require('babel-polyfill');
```

Comment:
What's the need for the polyfill?

Reply:
@omgimalexis it needed for the async

Reply:
According to [babel docs](https://babeljs.io/docs/en/babel-polyfill#usage-in-node-browserify-webpack) the proper way to add the polyfills is using `.babelrc`:
https://github.com/pymedusa/Medusa/blob/2d21ecd593d184f8fa5e45e07ae128e9ebc6a9b4/themes-default/slim/.babelrc#L3
change to:
```json
    ["@babel/preset-env", {
      "useBuiltIns": "usage"
    }]
```

Keep in mind that adding the polyfills is adding a big chunk of code, that might not actually be needed by the browsers we support...

Reply:
honestly i'm still confused why I can use await in vue files but not js?

Reply:
> honestly i'm still confused why I can use await in vue files but not js?

No vue **files** currently use `async`/`await`. This is an important distinction, because vue & js files from the `themes-default/slim/src/` folder are transpiled using babel.
The places that currently use async/await are vue components in **mako** files.

```js
// themes-default/slim/views/addShows_addExistingShow.mako
  177,30:             const response = await apiRoute.post('addShows/addExistingShows', formData);

// themes-default/slim/views/addShows_newShow.mako
  291,30:             const response = await apiRoute.post('addShows/addNewShow', formData);
  378,34:                 const response = await api.get('internal/searchIndexersForShowName', config);

// themes-default/slim/views/config_notifications.mako
  356,30:             const response = await apiRoute('home/loadShowNotifyLists')
  364,30:             const response = await apiRoute('home/loadShowNotifyLists')
  381,30:             const response = await apiRoute('home/getPushbulletDevices', { params: { api: pushbulletApiKey }});
  406,30:             const response = await apiRoute('home/testPushbullet', { params: { api: pushbulletApiKey }});
  421,30:             const response = await apiRoute('home/testJoin', { params: { api: joinApiKey }});
  431,30:             const response = await apiRoute('home/twitterStep1');
  442,34:                 const response = await apiRoute('home/twitterStep2', { params: { key: twitter.key } });
  451,34:                 const response = await apiRoute('home/testTwitter');

// themes-default/slim/views/config_search.mako
  319,26:             const resp = await apiRoute.get('home/testTorrent', { params });
  337,26:             const resp = await apiRoute.get('home/testNZBget', { params });
  355,26:             const resp = await apiRoute.get('home/testSABnzbd', { params });

// themes-default/slim/views/vue-components/quality-chooser.mako
  154,30:             const response = await api.get(url);
  189,30:             const response = await api.post(url, { type: 'ARCHIVE_EPISODES' });
```

Basically what this means is the browsers support `async`/`await` natively, but if those components were converted to `vue` SFCs right now, you would hit the exact same error from babel.

That said, you can choose to disable the babel plugins that transpile `async` functions.
I'm **not sure how good** of a solution this is or if it's even recommended.

But change:
https://github.com/pymedusa/Medusa/blob/2d21ecd593d184f8fa5e45e07ae128e9ebc6a9b4/themes-default/slim/.babelrc#L3
To:
```json
    ["@babel/preset-env", {
      "exclude": [
        "transform-regenerator",
        "transform-async-to-generator"
      ]
    }]
```

Reply:
Tnx a million for the detailed explanation. I really appreciate it. Especially now that we're kinda dead in the water regarding the entire frontend. So i'm tryig to make the best of it.